### PR TITLE
feat: add ctest/1.0.0

### PR DESCRIPTION
**Pull request by gocarlos**
- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/ctest/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import os
+from conans import CMake, ConanFile, tools
```
Comment (prince-chrismc) [inline recipes/ctest/all/conanfile.py:2]:
```suggestion
from conans import ConanFile, tools
```

File: recipes/ctest/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    topics = ("conan", "testing", "testing-framework", "unit-testing")
+    settings = "os"
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
```
Comment (prince-chrismc) [inline recipes/ctest/all/conanfile.py:20]: Not used

File: recipes/ctest/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+
+class CtestConan(ConanFile):
+    name = "ctest"
+    license = "Apache-2.0"
+    homepage = "https://github.com/bvdberg/ctest"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "ctest is a unit test framework for software written in C."
+    topics = ("conan", "testing", "testing-framework", "unit-testing")
+    settings = "os"
```
Comment (prince-chrismc) [inline recipes/ctest/all/conanfile.py:12]:
```suggestion
```

**Comment by SSE4** [comment]
oh, this one is confusing. I was thinking of [ctest](https://cmake.org/cmake/help/latest/manual/ctest.1.html) from the CMake distribution. may we add author prefix?

**Comment by Croydon** [comment]
> oh, this one is confusing. I was thinking of ctest from the CMake distribution. may we add author prefix?

I had the same though. 👍  for bvdberg-ctest

**Comment by uilianries** [comment]
I agree, people can be confused due CMake project.

**Comment by gocarlos** [comment]
something we should do to the recipe to say that this currently does not work with cpp?

ref https://github.com/bvdberg/ctest/pull/41

**Comment by SSE4** [comment]
@danimtb seems like we have some problem with CI here

**Comment by danimtb** [comment]
I am relaunching the CI here to debug this issue. Thanks

File: recipes/bvdberg-ctest/all/test_package/test_package.c [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+		printf("\nNOTE: all tests passed, it works! ;)\n");
+	}
+
+	return result;
+}
+
+CTEST(suite1, test2)
+{
+	ASSERT_EQUAL(1, 1);
+}
```
Comment (danimtb) [inline recipes/bvdberg-ctest/all/test_package/test_package.c:22]: To pass hook checks
```suggestion
}

```

**Pull request merged**